### ISSUE TITLE
All MFEE scenarios fail due to no route available message from PFS

### ISSUE THREAD

**Issue by agatsoh** [raiden-network/raiden#5400]
## Problem Definition

All the mfee scenarios fail due to the same reason of no route available message by the PFS

Analysing MFEE1 scenario:

Scenario in a nutshell:
node0: `0x45304c98dFA48c61bC6231BCd7e1d355a4f4b3b6`
node1: `0x029fFC9FEFFc1ba0Ff44d242767424FB990284Ed`
node2: `0x6fdbA48e2693AdCAfAD20a9E93d4a1Cc0e93f3cc`
node3: `0xfc65E6AbF01A42F0bF71a4ce9e9cA2FA624b77C5`

Scenario logs
```
{
  "task": "<TransferTask: {'from': 0, 'to': 3, 'amount': 10000, 'expected_http_status': 200}>",
  "event": "Task errored",
  "logger": "scenario_player.tasks.base",
  "level": "error",
  "timestamp": "2019-11-28 10:29:42.513623",
  "exception": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/base.py\", line 74, in __call__\n    return self._run(*args, **kwargs)\n  File \"/usr/local/lib/python3.7/site-packages/scenario_player/tasks/api_base.py\", line 59, in _run\n    f'HTTP status code \"{resp.status_code}\" while fetching {url}. '\nscenario_player.exceptions.legacy.RESTAPIStatusMismatchError: HTTP status code \"409\" while fetching http://127.0.0.1:57961/api/v1/payments/0x62083c80353Df771426D209eF578619EE68D5C7A/0xfc65E6AbF01A42F0bF71a4ce9e9cA2FA624b77C5. Expected 200: {\"errors\": \"Payment couldn't be completed because: there is no route available\"}"
}
```

What does the exception say:

1. `Payment couldn't be completed because: there is no route available`

2. Occurs at transfer task
`- transfer: {from: 0, to: 3, amount: 10_000, expected_http_status: 200}`

line 57

3. while fetching routes request
`http://127.0.0.1:57961/api/v1/payments/0x62083c80353Df771426D209eF578619EE68D5C7A/0xfc65E6AbF01A42F0bF71a4ce9e9cA2FA624b77C5`
where

`0x62083c80353Df771426D209eF578619EE68D5C7A` : token address
`0xfc65E6AbF01A42F0bF71a4ce9e9cA2FA624b77C5` : node3 address

[mfee1_flat_fee_2019_11_28.tar.gz](https://github.com/raiden-network/raiden/files/3901946/mfee1_flat_fee_2019_11_28.tar.gz)

[mfee2_proportional_fees_2019_11_28.tar.gz](https://github.com/raiden-network/raiden/files/3901953/mfee2_proportional_fees_2019_11_28.tar.gz)

**Comment by karlb** [comment]
Node0 has sent a PFSCapacity update:
```json
{
  "current_user": "@0x45304c98dfa48c61bc6231bcd7e1d355a4f4b3b6:transport02.raiden.network",
  "node": "0x45304c98dFA48c61bC6231BCd7e1d355a4f4b3b6",
  "transport_uuid": "df31536d-9fec-485e-a638-06ac2a2d243e",
  "room_name": "raiden_goerli_path_finding",
  "room": "<Room id='!yAaahHVZHRfbWQhpoi:transport01.raiden.network' alias='#raiden_goerli_path_finding:transport01.raiden.network'>",
  "data": "{\"other_participant\": \"0x029fFC9FEFFc1ba0Ff44d242767424FB990284Ed\", \"reveal_timeout\": \"50\", \"other_capacity\": \"0\", \"canonical_identifier\": {\"channel_identifier\": \"3788\", \"token_network_address\": \"0xDa1fBc048f503635950058953f5c60FC1F564ee6\", \"chain_identifier\": \"5\"}, \"updating_capacity\": \"100000\", \"updating_nonce\": \"0\", \"signature\": \"0xabab0898bdaa98f75223c784a049f3ad160d4853795067d86665bd02c4d0f0522b8b4e7c002af7ba666b9844baaece4ff6b34cf695fe3bab472b2ae836f89f7b1b\", \"other_nonce\": \"0\", \"updating_participant\": \"0x45304c98dFA48c61bC6231BCd7e1d355a4f4b3b6\", \"type\": \"PFSCapacityUpdate\"}\\n{\"fee_schedule\": {\"cap_fees\": true, \"proportional\": \"0\", \"imbalance_penalty\": null, \"flat\": \"50\"}, \"canonical_identifier\": {\"channel_identifier\": \"3788\", \"token_network_address\": \"0xDa1fBc048f503635950058953f5c60FC1F564ee6\", \"chain_identifier\": \"5\"}, \"signature\": \"0xd72bb612944eac04709e532667b568e5e6614c92a8f832a850bdd2978fdbd5f51542eca30449859ffe7bfb9a3392da026df6e11f879d8b4c9bd4f26564785dac1b\", \"timestamp\": \"2019-11-28T10:29:07.338792\", \"updating_participant\": \"0x45304c98dFA48c61bC6231BCd7e1d355a4f4b3b6\", \"type\": \"PFSFeeUpdate\"}",
  "event": "Broadcast",
  "logger": "raiden.network.transport.matrix.transport",
  "level": "debug",
  "greenlet_name": "MatrixTransport._run node:0x45304c98dFA48c61bC6231BCd7e1d355a4f4b3b6",
  "timestamp": "2019-11-28 10:29:07.353562"
}
```
but that has never reached or never been processed by the PFS:
```sh
journalctl CONTAINER_NAME=dev_pfs-goerli-with-fee_1 > all.log
cat all.log | perl -pe 's/^.*?\{/{/' | grep { | grep Capacity | grep -i 0x45304c98dFA48c61bC6231BCd7e1d355a4f4b3b6 | jq .
# no output
```

**Comment by karlb** [comment]
The message successfully arrived in the matrix room:
```
{
  "content": {
    "body": "{\"other_participant\": \"0x029fFC9FEFFc1ba0Ff44d242767424FB990284Ed\", \"reveal_timeout\": \"50\", \"other_capacity\": \"0\", \"canonical_identifier\": {\"channel_identifier\": \"3788\", \"token_network_address\": \"0xDa1fBc048f503635950058953f5c60FC1F564ee6\", \"chain_identifier\": \"5\"}, \"updating_capacity\": \"100000\", \"updating_nonce\": \"0\", \"signature\": \"0xabab0898bdaa98f75223c784a049f3ad160d4853795067d86665bd02c4d0f0522b8b4e7c002af7ba666b9844baaece4ff6b34cf695fe3bab472b2ae836f89f7b1b\", \"other_nonce\": \"0\", \"updating_participant\": \"0x45304c98dFA48c61bC6231BCd7e1d355a4f4b3b6\", \"type\": \"PFSCapacityUpdate\"}\n{\"fee_schedule\": {\"cap_fees\": true, \"proportional\": \"0\", \"imbalance_penalty\": null, \"flat\": \"50\"}, \"canonical_identifier\": {\"channel_identifier\": \"3788\", \"token_network_address\": \"0xDa1fBc048f503635950058953f5c60FC1F564ee6\", \"chain_identifier\": \"5\"}, \"signature\": \"0xd72bb612944eac04709e532667b568e5e6614c92a8f832a850bdd2978fdbd5f51542eca30449859ffe7bfb9a3392da026df6e11f879d8b4c9bd4f26564785dac1b\", \"timestamp\": \"2019-11-28T10:29:07.338792\", \"updating_participant\": \"0x45304c98dFA48c61bC6231BCd7e1d355a4f4b3b6\", \"type\": \"PFSFeeUpdate\"}",
    "msgtype": "m.text"
  },
  "origin_server_ts": 1574936947992,
  "room_id": "!yAaahHVZHRfbWQhpoi:transport01.raiden.network",
  "sender": "@0x45304c98dfa48c61bc6231bcd7e1d355a4f4b3b6:transport02.raiden.network",
  "type": "m.room.message",
  "unsigned": {
    "age": 104619746,
    "m.relations": {
      "m.annotation": {
        "chunk": [
          {
            "type": "m.reaction",
            "key": "👀",
            "count": 1
          }
        ]
      }
    }
  },
  "event_id": "$lrTpsqavrlX60-ixszeOI0Dg8-SZuunDZUREKE-izVE",
  "user_id": "@0x45304c98dfa48c61bc6231bcd7e1d355a4f4b3b6:transport02.raiden.network",
  "age": 104619746
}
```
So either the PFS's matrix code does not fetch the message properly, or the PFS silently discards the message.

**Comment by Dominik1999** [comment]
Closed and merged into META Issue #5433 and by #673
